src/small.rs

```diff
@@ -219,23 +469,19 @@ fn batch_exp(bases: &mut [G1Affine], coeff: Fr) {
                 for (base, products) in bases.iter_mut().zip(products.iter_mut()) {
                     *products = wnaf.base(base.into_projective(), 1).scalar(coeff);
                 }
+                // Normalize the projective products.
```

Comment:
(repeated comment) I believe the intention of this code is to isolate tasks based upon their CPU requirement (i.e. use all cores for scalar multiplication, then do normalization on all cores, then do the conversion). I think that we should maintain the separation between w-NAF and projective->affine conversion.

Calling `.into_affine()` after you've normalized the projective points is essentially a free operation, you are just moving the x and y coordinates into a new structure I believe. So maybe those two operations can be a single rayon operation.